pkg/auth/auth.go

```diff
@@ -0,0 +1,80 @@
+package auth
+
+import (
+	"os"
+	"strings"
+
+	"github.com/cli/go-gh/internal/set"
+	"github.com/cli/go-gh/pkg/config"
+)
+
+const (
+	defaultHost           = "github.com"
+	ghEnterpriseToken     = "GH_ENTERPRISE_TOKEN"
+	ghHost                = "GH_HOST"
+	ghToken               = "GH_TOKEN"
+	githubEnterpriseToken = "GITHUB_ENTERPRISE_TOKEN"
+	githubToken           = "GITHUB_TOKEN"
+	oauthToken            = "oauth_token"
+	hostsKey              = "hosts"
+)
+
+func TokenForHost(cfg *config.Config, host string) (string, string) {
```

Comment:
Yeah I think I agree, these functions feel like they should know how to access the config themselves, I can't think of a scenario where we would want the user to pass in their own `config.Config` instance. My only worry is that `config.Config` becomes an implicit dependency of these functions rather than an explicit one.